src/components/LinterMessage/index.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import { Alert } from 'react-bootstrap';
+
+import styles from './styles.module.scss';
+import { LinterMessage as LinterMessageType } from '../../linter';
+
+export type PublicProps = {
+  description: LinterMessageType['description'];
+  message: LinterMessageType['message'];
+  type: LinterMessageType['type'];
+};
+
+const getAlertVariant = (type: PublicProps['type']) => {
+  switch (type) {
+    case 'error':
+      return 'danger';
+    case 'warning':
+      return 'warning';
+    default:
+    case 'notice':
+      return 'secondary';
+  }
+};
+
+const urlPattern = /^https?:\/\//;
```

Comment:
Why is this extracted outside the function? It's used once, so I think we can inline it.

Reply:
It's an old Python habit of only compiling the regex once but who knows how JS works. I'll move it inline.